woocommerce/woocommerce-ios
# Lightweight storefront: Pages selector in Themes Preview

Closes #11471, #11472

## Description


This PR updates the Themes Preview view with the Pages selector UI:
| Pages selector shown at the top | Page selection sheet shown when tapped | "Preview" text shown if no Pages found |
|-|-|-|
| ![Simulator Screenshot - iPhone 14 Pro - 2023-12-19 at 12 07 25](https://github.com/woocommerce/woocommerce-ios/assets/266376/e59ee45d-82c7-456b-8d70-688f900067cf)  | ![Simulator Screenshot - iPhone 14 Pro - 2023-12-19 at 12 07 30](https://github.com/woocommerce/woocommerce-ios/assets/266376/f6b4916a-86ec-45fa-8a30-a76c510ba721) | ![Simulator Screenshot - iPhone 14 Pro - 2023-12-19 at 12 07 46](https://github.com/woocommerce/woocommerce-ios/assets/266376/b81f6b60-c539-491e-9012-91d32c2ee85c) |

Things that will be handled in a separate PR:

1.  **the design of the page selector area is still subject to change** so it doesn't match the Figma design. The Figma design does not properly match the available space in an iOS toolbar.
2. There is no loading state for when a new page is currently being loaded in the webview, so the transition might look jarring. I attempted it for this PR but it's more complex than I thought (suggestions welcome!)

## Testing instructions


1. Start with a WooExpress site,
3. Go to Menu > Themes
4. Pick a theme
5. If you picked Attar, it should only show "Preview",
6. If you picker another theme, it should show the Page selector like in the design above,
7. Tap the selector, pick a page,
8. Ensure the page is loaded in the webview (check the content of the webpage)
9. Ensure that the selected page title is now shown on the Page selector

## Video

https://github.com/woocommerce/woocommerce-ios/assets/266376/28f8c26d-aefc-43d1-a687-665e1f58d7d0
